tlslite/tlsconnection.py

```diff
@@ -1995,37 +1995,40 @@ def _serverSendTickets(self, settings):
         if not settings.ticketKeys:
             return
 
-        # prepare the ticket
-        ticket = SessionTicketPayload()
-        ticket.create(self.session.resumptionMasterSecret,
-                      self.version,
-                      self.session.cipherSuite,
-                      int(time.time()),
-                      getRandomBytes(len(settings.ticketKeys[0])))
-
-        # encrypt the ticket
-        if settings.ticketCipher in ("aes128gcm", "aes256gcm"):
-            cipher = createAESGCM(settings.ticketKeys[0],
-                                  settings.cipherImplementations)
-        else:  # assume chacha, enforced by handshake settings
-            cipher = createCHACHA20(settings.ticketKeys[0],
-                                    settings.cipherImplementations)
-
-        # all AEADs we support require 12 byte nonces/IVs
-        iv = getRandomBytes(12)
-
-        encrypted_ticket = cipher.seal(iv, ticket.write(), b'')
+        for _ in range(settings.ticket_count):
+            # prepare the ticket
+            ticket = SessionTicketPayload()
+            ticket.create(self.session.resumptionMasterSecret,
+                          self.version,
+                          self.session.cipherSuite,
+                          int(time.time()),
+                          getRandomBytes(len(settings.ticketKeys[0])))
+
+            # encrypt the ticket
+            if settings.ticketCipher in ("aes128gcm", "aes256gcm"):
+                cipher = createAESGCM(settings.ticketKeys[0],
```

Comment:
Unrelated to this PR, but these choices seem quite risky for session tickets due to their nonce misuse properties

Reply:
you mean static key with random IV?

Reply:
Yes. There is very little you can do prevent a nonce re-use.